datajunction-clients/python/README.md

```diff
@@ -16,130 +24,346 @@ from datajunction import DJClient
+        filters: Optional[List[str]],
+        engine_name: Optional[str],
+        engine_version: Optional[str])
+  - data( metrics: List[str],
+        dimensions: Optional[List[str]],
+        filters: Optional[List[str]],
+        engine_name: Optional[str],
+        engine_version: Optional[str],
+        async_: bool = True)
+
+## DJ Builder : Data Modelling
+
+In this section we'll show you few examples to modify the DJ data model and its nodes.
+
+### Start Here
+
+To initialize the DJ builder:
 
 ```python
-from datajunction import Engine
+from datajunction import DJBuilder
 
-engine = Engine(
-    name="spark",
-    version="3.2.2",
-    uri="..."
-)
-engine.publish()
+djb = DJBuilder("http://localhost:8000")
 ```
 
-To attach an engine to a catalog:
+**NOTE**
+If you are running in our demo docker container please change the above URL to "http://dj:8000".
+
+### Namespaces
+
+To access a namespace or check if it exists you can use the same simple call:
+
 ```python
-catalog.add_engine(engine)
+djb.namespace("default")
+
+Namespace(dj_client=..., namespace='default')
 ```
+```python
+djb.namespace("foo")
+
+[DJClientException]: Namespace `foo` does not exist.
+```
+
+To create a namespace:
+
+```python
+djb.create_namespace("foo")
+
+Namespace(dj_client=..., namespace='foo')
+```
+
+To delete (or restore) a namespace:
+
+```python
+djb.delete_namespace("foo")
+
+djb.restore_namespace("foo")
+```
+
+**NOTE:**
+The `cascade` parameter in both of above methods allows for cascading
+effect applied to all underlying nodes and namespaces. Use it with caution!
 
 ### Nodes
 
+To learn what **Node** means in the context of DJ, please check out [this datajuntion.io page](https://datajunction.io/docs/0.1.0/dj-concepts/nodes/).
+
+To list all (or some) nodes in the system you can use the `list_<node-type>()` methods described
+in the **DJ Client : Basic Access** section or you can use the namespace based method:
+
 All nodes for a given namespace can be found with:
 ```python
-dj.namespace("default").nodes()
+djb.namespace("default").nodes()
 ```
 
 Specific node types can be retrieved with:
 ```python
-dj.namespace("default").sources()
-dj.namespace("default").dimensions()
-dj.namespace("default").metrics()
-dj.namespace("default").transforms()
-dj.namespace("default").cubes()
+djb.namespace("default").sources()
+djb.namespace("default").dimensions()
+djb.namespace("default").metrics()
+djb.namespace("default").transforms()
+djb.namespace("default").cubes()
 ```
 
 To create a source node:
 
 ```python
-from datajunction import NodeMode
-
-repair_orders = dj.new_source(
+repair_orders = djb.create_source(
     name="repair_orders",
     display_name="Repair Orders",
     description="Repair orders",
     catalog="dj",
     schema_="roads",
     table="repair_orders",
 )
-repair_orders.save(mode=NodeMode.PUBLISHED)
 ```
 
-Nodes can also be created as drafts with:
+Nodes can also be created in draft mode:
```

Comment:
Would it make sense to show this creation path instead?
```
repair_orders = djb.create_source(
    name="repair_orders",
    display_name="Repair Orders",
    description="Repair orders",
    catalog="dj",
    schema_="roads",
    table="repair_orders",
    mode=NodeMode.DRAFT,
)
```

I ask because otherwise the creator will first create the node in published mode (which won't work for nodes with non-existent parents) and then switch it to draft, whereas the former just immediately creates it as a draft. Alternatively we could mention that there are two ways to create a node, one using `create_source` and one with creating the Node object locally. Or we could just leave this to the datajunction.io docs...

Reply:
Good point @shangyian . I'll change it to:
```
```python
from datajunction import NodeMode

repair_orders = djb.create_source(
    name="repair_orders",
    display_name="Repair Orders",
    description="Repair orders",
    catalog="dj",
    schema_="roads",
    table="repair_orders",
    mode=NodeMode.DRAFT
)
```